loadbalancer/node.go

```diff
@@ -26,3 +38,28 @@ func (upstream node) HealthCheck(ctx context.Context) error {
 
 	return c.Close()
 }
+
+var ranges = []struct {
+	min, max time.Duration
+}{
+	{0, time.Millisecond},
+	{time.Millisecond, 10 * time.Millisecond},
+	{10 * time.Millisecond, 100 * time.Millisecond},
+	{100 * time.Millisecond, 1 * time.Second},
+	{1 * time.Second, math.MaxInt64},
+}
+
+func calcTier(err error, elapsed time.Duration) (upstream.Tier, error) {
+	if err != nil {
+		// preserve old tier
+		return -1, err
+	}
+
+	for i := range ranges {
+		if elapsed >= ranges[i].min && elapsed < ranges[i].max {
```

Comment:
more of a nit, but we could simplify if we drop `.max` field, and just start the search from the end of the list towards the beginning

if `elapsed` >= `tier.min`, it's the tier.